runtime/sema/gen/main.go

```diff
@@ -588,9 +589,20 @@ func (g *generator) VisitEntitlementDeclaration(decl *ast.EntitlementDeclaration
 	return
 }
 
-func (*generator) VisitEntitlementMappingDeclaration(_ *ast.EntitlementMappingDeclaration) struct{} {
-	// TODO
-	panic("entitlement declarations are not supported")
+func (g *generator) VisitEntitlementMappingDeclaration(decl *ast.EntitlementMappingDeclaration) (_ struct{}) {
+
+	entitlementMappingName := decl.Identifier.Identifier
+	typeVarName := typeVarName(entitlementMappingName)
+	typeVarDecl := entitlementMapTypeLiteral(entitlementMappingName, decl.Associations)
```

Comment:
Once https://github.com/onflow/cadence/issues/2643 is implemented, this should be updated to include support for includes

Reply:
Will add this in a follow-up PR, as it requires additional work in the type code generator